quinn-h3/src/proto/connection.rs

```diff
@@ -226,8 +226,8 @@ impl Connection {
         );
     }
 
-    pub fn requests_in_flight(&self) -> usize {
-        self.requests_in_flight.len()
+    pub fn shutdown_complete(&self) -> bool {
```

Comment:
If there's only one user, is it really useful for this to be a separate method?

Reply:
The idea here is to move non-async logic into `proto` module. There is still too much in `src/connection.rs` in my opinion.

Reply:
OK.